### PR TITLE
Add `.bot` and `.idx` files to EK60/EK80 parsing

### PR DESCRIPTION
**Pull request by ctuguinay**
Addresses #1315

### CODE REVIEW

**Comment by ctuguinay** [comment]
@leewujung This should also be ready to look at. Only recent change I made was the `**kwarg` thing, Not sure why it still says read the docs pending. The test docs completed already.

File: echopype/convert/api.py
Comment (leewujung) [inline echopype/convert/api.py:318]: This is super tiny but i think since the backtick is really used for docstring, maybe we can skip it here.
```suggestion
                f"There is no file named {bot_file}. The .BOT file must be contained in "
                + " the same directory as that of the input 'raw' file."
```

File: echopype/convert/api.py
Comment (leewujung) [inline echopype/convert/api.py:330]:
```suggestion
                f"There is no file named {idx_file}. The .IDX file must be contained in "
                + " the same directory as that of the input 'raw' file."
```

File: echopype/convert/set_groups_base.py
Comment (leewujung) [inline echopype/convert/set_groups_base.py:419]: Same minor thing: remove all backticks from the comment here and below?
```suggestion
                "long_name": "Timestamps from .IDX datagrams",
```

File: echopype/convert/set_groups_base.py
Comment (leewujung) [inline echopype/convert/set_groups_base.py:436]:
```suggestion
                "comment": "Data derived from `.IDX` datagrams. Aligns time-wise with this "
                + "dataset's `time3` dimension. "
                + "This is different from latitude stored in the NMEA data gram.",
```

File: echopype/convert/set_groups_base.py
Comment (leewujung) [inline echopype/convert/set_groups_base.py:443]:
```suggestion
                "comment": "Data from the IDX datagrams. Aligns time-wise with this "
                + "dataset's `time3` dimension. "
                + "This is different from longitude from the NMEA datagram.",
```

File: echopype/convert/set_groups_base.py
Comment (leewujung) [inline echopype/convert/set_groups_base.py:491]:
```suggestion
                "long_name": "Timestamps from the BOT datagrams",
```

File: echopype/convert/set_groups_base.py
Comment (leewujung) [inline echopype/convert/set_groups_base.py:498]:
```suggestion
            {"long_name": "Echosounder detected seafloor depth from the BOT datagrams."}
```

File: echopype/convert/set_groups_ek60.py
Comment (leewujung) [inline echopype/convert/set_groups_ek60.py:320]: I think this is fine, but I am curious about if the later checks (those involving `self.parser_obj.idx`) are included to catch cases where `self.parser_obj.idx_file != ""` is satisfied but `self.parser_obj.idx` does not contain any data?

File: echopype/tests/convert/test_convert_azfp.py
Comment (leewujung) [inline echopype/tests/convert/test_convert_azfp.py:199]: Seems like this line is ok without the additions?

File: echopype/tests/convert/test_convert_ek.py
Comment (leewujung) [inline echopype/tests/convert/test_convert_ek.py:51]:
```suggestion
def test_convert_ek_with_bot_file(file, sonar_model):
```

File: echopype/tests/convert/test_convert_ek.py
Comment (leewujung) [inline echopype/tests/convert/test_convert_ek.py:97]:
```suggestion
def test_convert_ek_with_idx_file(file, sonar_model):
```

**Review by leewujung — APPROVED** [review]
@ctuguinay : Thanks for the PRs, looks great! I only have cosmetic suggestions.

File: echopype/convert/set_groups_base.py
Comment (ctuguinay) [reply echopype/convert/set_groups_base.py:436]: Ah yeah I agree. Good catch

File: echopype/convert/set_groups_ek60.py
Comment (ctuguinay) [reply echopype/convert/set_groups_ek60.py:320]: Yup, that's what it's meant to catch. In the case where the parser reads nothing from the IDX file

File: echopype/tests/convert/test_convert_azfp.py
Comment (ctuguinay) [reply echopype/tests/convert/test_convert_azfp.py]: Ah yeah I removed this. Thanks!

File: echopype/convert/set_groups_base.py
Comment (leewujung) [inline echopype/convert/set_groups_base.py:415]: Sorry I should have noticed this yesterday... I just checked the IDX0 datagram content. I think we should add `ping_number` and `filer_offset` to the stored data as well, since in the manual there are some intricacies in how the pings are associated:
https://www.simrad.online/ek80/interface/ek80_interface_en_a4.pdf#page=162

Also, to make it easy to recognize these, how about prefixing the original data variable names with `idx_`, so there is no confusion of which one is which? i.e. `idx_distance`, `idx_latitude`, etc.

**Comment by ctuguinay** [comment]
Pinned `numpy<2` and made the IDX prefix change. I'll read up on the datagrams

**Comment by ctuguinay** [comment]
Added ping number and file offset to values that are parsed in the datagram and added to the group. I still don't completely understand file offset...so I'll figure that out first before I add their corresponding group attributes. I've also never used the replay bar in Echoview so I'll try that out tomorrow.

**Comment by leewujung** [comment]
I think for the replay thing they are referring to the EK80 software. The windows machine also has that installed.

**Comment by ctuguinay** [comment]
@leewujung Ok this is ready for review again haha. Let's figure out #1344 another time, I left a small TODO in the code as a reminder for that.

File: echopype/convert/set_groups_base.py
Comment (leewujung) [inline echopype/convert/set_groups_base.py:413]: Just to make it absolutely clear where these came from
```suggestion
                "idx_ping_number": xr.DataArray(
                    np.array(self.parser_obj.idx["ping_number"]),
                    dims=("time3"),
                    coords={"time3": timestamp_array},
                ),
                "idx_file_offset": xr.DataArray(
                    np.array(self.parser_obj.idx["file_offset"]),
                    dims=("time3"),
                    coords={"time3": timestamp_array},
                ),
                "idx_vessel_distance": xr.DataArray(
                    np.array(self.parser_obj.idx["vessel_distance"]),
                    dims=("time3"),
                    coords={"time3": timestamp_array},
                ),
```

Comment (leewujung) [reply echopype/convert/set_groups_base.py:413]: And just because I am looking at this... why wouldn't you do the attribute assignment here in each of the xr.DataArray too?

File: echopype/convert/parse_base.py
Comment (leewujung) [inline echopype/convert/parse_base.py:570]: Since these are already under `self.idx` you could remove the idx_ here also.

**Review by leewujung — DISMISSED** [review]
@ctuguinay : The changes look great! I only have very nitpicking comments. Feel free to merge!

Also I wonder what the difference between the 2 EK60 test files (or between the 2 test EK80 files) are?

File: echopype/convert/set_groups_base.py
Comment (ctuguinay) [reply echopype/convert/set_groups_base.py:413]: Ah yeah you're right I'll do that

File: echopype/convert/parse_base.py
Comment (ctuguinay) [reply echopype/convert/parse_base.py:570]: made this change 👍

**Comment by ctuguinay** [comment]
> Also I wonder what the difference between the 2 EK60 test files (or between the 2 test EK80 files) are?

No big difference. I just wanted to make sure I had two of each, since I like the number 2 😄

**Comment by ctuguinay** [comment]
@leewujung 😭 ok it now requires another review approval before I can merge since I made the changes you suggested above
image

**Review by leewujung — APPROVED** [review]
🤯 ok here you go!

**Pull request merged**
